BE/src/main/java/baseball/service/GameService.java

```diff
@@ -0,0 +1,78 @@
+package baseball.service;
+
+import baseball.domain.Game;
+import baseball.domain.Score;
+import baseball.domain.Team;
+import baseball.exception.GameNotFoundException;
+import baseball.exception.TeamNotFoundException;
+import baseball.repository.GameRepository;
+import baseball.repository.TeamRepository;
+import baseball.service.dto.GameDTO;
+import baseball.service.dto.GameScoreDTO;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class GameService {
+
+    private static final int NUMBER_OF_TEAM = 10;
+
+    private GameRepository gameRepository;
+    private TeamRepository teamRepository;
+
+    public GameService(GameRepository gameRepository, TeamRepository teamRepository) {
+        this.gameRepository = gameRepository;
+        this.teamRepository = teamRepository;
+    }
+
+    public List<GameDTO> getGameDTOList() {
+        if (gameRepository.count() != 5) {
+            saveGames();
+        }
+
+        List<GameDTO> gameDTOs = new ArrayList<>();
+        Iterable<Game> games = gameRepository.findAll();
+
+        for (Game game : games) {
+            Long id = game.getId();
+            Team homeTeam = teamRepository.findById(game.getHomeTeamId()).orElseThrow(TeamNotFoundException::new);
+            Team awayTeam = teamRepository.findById(game.getAwayTeamId()).orElseThrow(TeamNotFoundException::new);
+
+            GameDTO gameDTO = new GameDTO(id, homeTeam.getName(), awayTeam.getName());
+            gameDTOs.add(gameDTO);
+        }
+        return gameDTOs;
```

Comment:
```suggestion
    return StreamSupport.stream(gameRepository.findAll().spliterator(), false)
        .map(game -> {
          Long id = game.getId();
          Team homeTeam = teamRepository.findById(game.getHomeTeamId()).orElseThrow(TeamNotFoundException::new);
          Team awayTeam = teamRepository.findById(game.getAwayTeamId()).orElseThrow(TeamNotFoundException::new);

          return new GameDTO(id, homeTeam.getName(), awayTeam.getName());
        })
        .collect(Collectors.toList());
```
Stream을 이용하면 이런식으로도 되겠죠?

Reply:
아직 스트림 사용이 익숙치 않은듯 합니다... GameDTO가 내부적으로 List<Game>을 갖도록 변경되어서 이부분도 현재는 필요하지 않아서 머리속에 저장하겠습니다~